### DISCUSSION TITLE
Backup Prior to Update

### DISCUSSION THREAD

**Discussion by shrimprugbysnowowl** [babybuddy/babybuddy#794] [Support]
I was running version 2.2.0 on a Synology NAS using Container Manager (Docker), and technically the linuxserver version, when I saw there was an update available.  Following the update and realizing that the old credentials no longer work, I've now realized that there is an [export-import procedure](https://docs.baby-buddy.net/import-export/#runserver) that I should have followed...  Is the data from the previous version lost?  I've read the other issues and discussion pertaining to potentially lost data but I'm not that experienced with Docker and I'm not following the exchange very well.  Thanks for any assistance that you can provide.

**Comment by cdubz** [comment]
@shrimprugbysnowowl can you share your docker compose configuration for the app? The data _should_ persist between updates but this might not happen if you haven't configured persistent storage for the Docker database.

**Reply by shrimprugbysnowowl** [reply]
> @shrimprugbysnowowl can you share your docker compose configuration for the app? The data _should_ persist between updates but this might not happen if you haven't configured persistent storage for the Docker database.

Good morning, @cdubz - we just experienced the error you've been tracking and working on in #893 (server error 500 while accessing dashboard).  Given my experience here with data not being retained when the container image is updated, I'm extremely hesitant to update the image from v2.2.1 to the latest release.  Did you have any comments on the docker compose config that I posted here back in May?  I'd be very grateful for any input you can provide.

**Reply by cdubz** [reply]
Hi, @shrimprugbysnowowl -- sorry for not getting back sooner. I haven't used Synology at all, so I'm not able to provide much support here.  What you essentially need to figure out is how to configure persistent storage for Baby Buddy's data. From your description it sounds like the database is not stored persistently so will be lost anytime the service or device is restarted. The Synology community forums may be able to provide better support for this, but I did a quick search there and didn't see anything obvious. There may be some Reddit threads on similar topics that could help, if you're a Reddit user.

**Reply by shrimprugbysnowowl** [reply]
Thank you for the reply - I'll do some digging.  I set the volume to save data to `/docker/babybuddy/data`, and I see an sqlite3 database file in the folder.  Is that where the data is stored primarily?

I'm also still able to access the admin panel.  I'm looking to export the data just in case my poking around clears it all again.  Per the [docs](https://docs.baby-buddy.net/import-export/), do I need to export each "core"/heading individually?  Or is there a more streamlined process?

**Reply by cdubz** [reply]
> Thank you for the reply - I'll do some digging. I set the volume to save data to /docker/babybuddy/data, and I see an sqlite3 database file in the folder. Is that where the data is stored primarily?

Yep! That's the important thing. If nothing else, you can grab that file before you try an update and if shit breaks again, just put your saved version of that file back there and it _should_ be all good.

The only other bit to consider is migrations... I'm not sure if this is handled automatically by LSIO's image (I don't think it is), but sometimes updates with have database changes that need to "miragte" your database. You'll need to figure out how to run the command `python manage.py migrate` inside the container after the upgrade.

> I'm also still able to access the admin panel. I'm looking to export the data just in case my poking around clears it all again. Per the [docs](https://docs.baby-buddy.net/import-export/), do I need to export each "core"/heading individually? Or is there a more streamlined process?

No streamlined process yet. I've got an old,old issue for that #124 😁

**Reply by shrimprugbysnowowl** [reply]
Thank you again.  I'll report back here for record in case anyone else encounters a similar issue.

**Reply by maxdd** [reply]
Would backuping the sqlite3 db be enough to recover lost data or deploy on a new machine?
Is that protected and hence would require the same admin account or something?
I feel like the export/import function is a little bit tedious becase one has to run it manually on all the "core" entries

**Comment by shrimprugbysnowowl** [comment]
Apologies for the delay, and thank you very much for the reply.

Synology's prior Docker implementation didn't use traditional compose/yml files. I hope the environment variables that I used in the setup will suffice:
PATH=/lsiopy/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PS1=$(whoami)@$(hostname):$(pwd)\$ 
HOME=/root
TERM=xterm
S6_CMD_WAIT_FOR_SERVICES_MAXTIME=0
S6_VERBOSITY=1
S6_STAGE2_HOOK=/init-hook
VIRTUAL_ENV=/lsiopy
LSIO_FIRST_PARTY=true

Then this is the location of the shared folder that it uses.
